java/ffi/README.md

```diff
@@ -0,0 +1,48 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Java FFI (C Data Interface)
+
+## Setup Build Environment
+
+install:
+ - Java 8 or later
+ - Maven 3.3 or later
+ - A C++11-enabled compiler
+ - CMake 3.11 or later
+ - Make or ninja build utilities
+
+## Building JNI wrapper shared library
+
+```
+mkdir -p ./target/build/
+pushd ./target/build/
+cmake ../..
+make
+popd
+```
+
+To use ninja, pass `-GNinja` when calling cmake and then use the `ninja` command instead of `make`.
+
+## Building and running tests
+
+```
+cd java
+mvn -Parrow-ffi install
```

Comment:
I changed it to be a separate build directory outside target so it should be fine now. I had to add a .gitignore file with that folder otherwise the license check fails.